server/report.go

```diff
@@ -0,0 +1,397 @@
+package main
```

Comment:
Can we split up these big files into separate ones? That would be easier to maintain I think.

Reply:
Done.